felix/bpf-gpl/tc.c

```diff
@@ -512,7 +512,13 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 			if (rt) {
 				flags = rt->flags;
 			}
-			if (!(flags & CALI_RT_IN_POOL) && !cali_rt_flags_local_host(flags)) {
+            // Don't perform SNAT if either:
+            // - packet is destined to an address in the same IP pool;
```

Comment:
```suggestion
            // - packet is destined to an address in a IP pool;
```
doesn't have to be the same, we do not even have the info that it is the same or not

Reply:
updated, thanks